main.go

```diff
@@ -14,128 +10,235 @@ const GLOBAL_NAME = "sqlparser"
-						return reject.Invoke(Error.New("the query references a table name with the wrong format"))
-					}
-					parts := strings.Split(node.Table.String(), "_")
-					if len(parts) < 2 {
-						return reject.Invoke(Error.New("table name isn't referencing the chain id"))
-					}
-					_, err = strconv.ParseInt(parts[len(parts)-1], 10, 64)
-					if err != nil {
-						return reject.Invoke(Error.New("parsing chain id in table name: " + err.Error()))
+			return resolve.Invoke(js.ValueOf(updated))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func validateTableName(this js.Value, args []js.Value) interface{} {
+	Error := js.Global().Get("Error")
+	Promise := js.Global().Get("Promise")
+	if len(args) < 1 {
+		return Promise.Call("reject", Error.New("missing required argument: tableName"))
+	}
+	tableName := args[0].String()
+	var isCreate bool
+	if len(args) == 2 && args[1].Type() == js.TypeBoolean {
+		isCreate = args[1].Bool()
+	}
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+		go func() interface{} {
+			table := &sqlparser.Table{Name: sqlparser.Identifier(tableName), IsTarget: true}
+			response := make(map[string]interface{})
+			if isCreate {
+				validTable, err := sqlparser.ValidateCreateTargetTable(table)
+				if err != nil {
+					return reject.Invoke(Error.New("error validating name: " + err.Error()))
+				}
+				response["prefix"] = validTable.Prefix()
+				response["chainId"] = validTable.ChainID()
+				response["name"] = validTable.Name()
+			} else {
+				validTable, err := sqlparser.ValidateTargetTable(table)
+				if err != nil {
+					return reject.Invoke(Error.New("error validating name: " + err.Error()))
+				}
+				response["prefix"] = validTable.Prefix()
+				response["chainId"] = validTable.ChainID()
+				response["tableId"] = validTable.TokenID()
+				response["name"] = validTable.Name()
+			}
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func getUniqueTableNames(this js.Value, args []js.Value) interface{} {
+	Error := js.Global().Get("Error")
+	Promise := js.Global().Get("Promise")
+	if len(args) < 1 {
+		return Promise.Call("reject", Error.New("missing required argument: statement"))
+	}
+	statement := args[0].String()
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+		go func() interface{} {
+			ast, err := sqlparser.Parse(statement)
+			if err != nil {
+				return reject.Invoke(Error.New("error parsing statement: " + err.Error()))
+			}
+			tableReferences := sqlparser.GetUniqueTableReferences(ast)
+			response := make([]interface{}, len(tableReferences))
+			for i := range tableReferences {
+				response[i] = tableReferences[i]
+			}
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func normalize(this js.Value, args []js.Value) interface{} {
```

Comment:
This function has been simplified. It does _less_ actual validation, and can be used to just normalize statements as much as possible. Invalid statements will still throw, but it no longer checks statement size, if the statement is empty, etc. You can use validateStatement for that, which essentially validates "in place".